### PR TITLE
add ETHOS transfer function

### PR DESCRIPTION
**Pull request by charlottenosam**
adding ETHOS transfer function to ps.c

### CODE REVIEW

**Comment by steven-murray** [comment]
Hi @charlottenosam it looks like the upstream branch might need to merge master into itself (and then onto your branch). Let me know if you need help with this!

File: src/py21cmfast/src/ps.c [hunk 1]
```diff
@@ -4257,12 +4257,46 @@ void FreeTsInterpolationTables(struct FlagOptions *flag_options) {
 
 
 
-double ETHOS_DAOs(struct AstroParams *astro_params_sfrd){
+double ETHOS_DAOs(double k, struct AstroParams *astro_params_sfrd){
```
[the remark this answers is not in the document]
Comment (charlottenosam) [reply src/py21cmfast/src/ps.c:4260]: no, I'll add it!

File: src/py21cmfast/src/ps.c [hunk 2]
```diff
@@ -4257,12 +4257,46 @@ void FreeTsInterpolationTables(struct FlagOptions *flag_options) {
...
 //this function returns the ratio between the power spectrum in an ETHOS model of hpeak and kpeak and that of LCDM. Depends on h_PEAK and k_PEAK which are in astro_params
 
 //nb: hpeak=astro_params_sfrd->h_PEAK
 //nb: kpeak=astro_params_sfrd->k_PEAK
 
-  return 1.0;
+    double b, d, tau, sig, c, h2, X, A, B, C;
```
[the remark this answers is not in the document]
Comment (charlottenosam) [reply src/py21cmfast/src/ps.c]: had to change some of them but it compiles now :)

**Comment by charlottenosam** [comment]
@steven-murray ok, I think this branch is up-to-date with master. 

I'm just trying to pull into the ETHOS_DM branch for now, not the master. Is possible for me to be a member contributor so I can work on this branch with @JulianBMunoz without having to do pull requests every time? (though also happy to carry on doing this if it keeps things tidier)

**Pull request merged**